builtInFunctions/baseAccountFreezer.go

```diff
@@ -0,0 +1,115 @@
+package builtInFunctions
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var guardianKeyPrefix = []byte(core.ElrondProtectedKeyPrefix + GuardiansKeyIdentifier)
+
+// BaseAccountFreezerArgs is a struct placeholder for
+// all necessary args to create a newBaseAccountFreezer
+type BaseAccountFreezerArgs struct {
+	Marshaller    marshal.Marshalizer
+	EpochNotifier vmcommon.EpochNotifier
+	FuncGasCost   uint64
+}
+
+type baseAccountFreezer struct {
+	marshaller marshal.Marshalizer
+
+	mutExecution sync.RWMutex
+	funcGasCost  uint64
+	currentEpoch uint32
+}
+
+func newBaseAccountFreezer(args BaseAccountFreezerArgs) (*baseAccountFreezer, error) {
+	if check.IfNil(args.Marshaller) {
+		return nil, ErrNilMarshaller
+	}
+	if check.IfNil(args.EpochNotifier) {
+		return nil, ErrNilEpochNotifier
+	}
+
+	return &baseAccountFreezer{
+		funcGasCost:  args.FuncGasCost,
+		marshaller:   args.Marshaller,
+		mutExecution: sync.RWMutex{},
+	}, nil
+}
+
+func (baf *baseAccountFreezer) checkArgs(
```

Comment:
This would look a bit "extra": func (baf ***baseAccountFreezer**) check**BaseAccountFreezer**Args, but understandable regarding "where it is being called from", so I'll change it as your suggestion